tools/generator-go-sdk/main.go

```diff
@@ -10,73 +10,102 @@ import (
-	apiServerEndpoint string
-	outputDirectory   string
-	services          []string
-	settings          generator.Settings
+	apiServerEndpoint   string
+	outputDirectoryPath string
+	sdkName             string
+	services            []string
+	settings            generator.Settings
+}
+
+type SDK struct {
+	// baseClientMethod is the go function for instantiating the particular base client
+	baseClientMethod string
+
+	// baseClientPackage is the go package where the base client can be found
+	baseClientPackage string
+
+	// clients is a map of Data API clients to use when retrieving definitions for the SDK
+	clients map[string]func(string) resourcemanager.Client
+
+	// commonModelsPackageNameFromVersion is a function that returns the go package name containing common types (models, constants, predicates), for the SDK version
+	commonModelsPackageNameFromVersion func(string) (*string, error)
+
+	// commonModelsPackagePathFromVersion is a function that returns the go import path where common types can be found, for the SDK version
+	commonModelsPackagePathFromVersion func(string) (*string, error)
+
+	// outputSubDirectory is the base directory where the SDK will reside in the go-azure-sdk repository
+	outputSubDirectory string
+
+	// versionMapper is a function that returns a local API version for a given version as returned by the Data API
+	versionMapper func(string) (*string, error)
+}
+
+var availableSDKs = map[string]SDK{
+	"resource-manager": {
+		baseClientMethod:  "NewResourceManagerClient",
+		baseClientPackage: "resourcemanager",
+		clients: map[string]func(string) resourcemanager.Client{
+			"resource-manager": resourcemanager.NewResourceManagerClient,
+		},
+		commonModelsPackageNameFromVersion: func(in string) (*string, error) {
+			return nil, nil
+		},
+		commonModelsPackagePathFromVersion: func(in string) (*string, error) {
+			return nil, nil
+		},
+		outputSubDirectory: "resource-manager",
+		versionMapper:      func(in string) (*string, error) { return &in, nil },
+	},
+
+	"microsoft-graph": {
+		baseClientMethod:  "NewMsGraphClient",
+		baseClientPackage: "msgraph",
+		clients: map[string]func(string) resourcemanager.Client{
+			"v1.0": resourcemanager.NewMicrosoftGraphStableV1Client,
+			"beta": resourcemanager.NewMicrosoftGraphBetaClient,
+		},
+		commonModelsPackageNameFromVersion: func(in string) (*string, error) {
+			return pointer.To("models"), nil
+		},
+		commonModelsPackagePathFromVersion: func(in string) (*string, error) {
+			path := fmt.Sprintf("common/%s/models", fmt.Sprintf(generator.GolangPackageNameForVersion(in)))
+			return &path, nil
+		},
+		outputSubDirectory: "microsoft-graph",
+		versionMapper: func(in string) (*string, error) {
+			version := ""
+			switch in {
+			case "StableV1":
+				version = "v1.0"
+			case "Beta":
+				version = "beta"
+			}
+			if version == "" {
+				return nil, fmt.Errorf("unsupported API version: %q", in)
+			}
+			return &version, nil
+		},
+	},
 }
 
 func main() {
```

Comment:
kinda outside the scope of this PR, but this is getting complex enough we should probably adopt `mitchellh/cli` here